src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -1031,6 +1028,31 @@ public void Sort(IComparer<T> comparer)
             return true;
         }
 
+        private void AddEnumerable(IEnumerable<T> enumerable)
+        {
+            Contract.Assert(enumerable != null);
+            Contract.Assert(!(enumerable is ICollection<T>), "We should have optimized for this beforehand.");
+
+            using (IEnumerator<T> en = enumerable.GetEnumerator())
+            {
+                _version++; // Even if the enumerable has no items, we can update _version.
```

Comment:
@justinvp It is just in case if this is called by another method later on, or e.g. the increment to `_version` is accidentally removed in `InsertRange`. It is just one write anyway, so I don't think it should cost much relative to all the other stuff going on in the method.